py/desispec/scripts/submit_prod.py

```diff
@@ -9,109 +9,302 @@
+            first_night = 20201214
+
+    if all_nights is None:
+        # all_nights = get_nights_in_date_range(first_night, last_night)
+        if verbose:
+            log.info("Populating all_nights with all of the nights with valid science "
+                     + f"exposures between {first_night} and {last_night} inclusive")
+        all_nights = get_all_valid_nights(first_night, last_night)
+    return sorted(all_nights)
+
 
-def submit_production(production_yaml, dry_run=False, error_if_not_available=False):
+def submit_production(production_yaml, queue_threshold=4800, dry_run_level=False):
     """
     Interprets a production_yaml file and submits the respective nights for processing
     within the defined production.
 
     Args:
         production_yaml (str): Pathname of the yaml file that defines the production.
-        dry_run (bool, optional): Default is False. Should the jobs written to the processing table actually be submitted
-            for processing.
-        error_if_not_available (bool, optional): Default is True. Raise as error if the required exposure table doesn't exist,
-            otherwise prints an error and returns.
+        queue_threshold (int): The number of jobs for the current user in the queue
+            at which the script stops submitting new jobs.
+        dry_run_level (int, optional): Default is 0. Should the jobs written to the processing table actually be submitted
+            for processing. This is passed directly to desi_proc_night.
 
     Returns:
         None.
     """
+    log = get_logger()
+    ## Load the yaml file
     if not os.path.exists(production_yaml):
-        raise IOError(f"Prod Yaml file doesn't exist: {production_yaml} not found. Exiting.")
-    conf = yaml.safe_load(open(production_yaml, 'rb'))
-    specprod = str(conf['name']).lower()
-    specprod = verify_variable_with_environment(var=specprod, var_name='specprod', env_name='SPECPROD')
-    if 'reservation' in conf:
-        reservation = str(conf['reservation'])
+        raise IOError(f"Prod yaml file doesn't exist: {production_yaml} not found.")
+    with open(production_yaml, 'rb') as yamlfile:
+        conf = yaml.safe_load(yamlfile)
+
+    ## Make sure the specprod matches, if not set it to that in the file
+    if 'SPECPROD' not in conf:
+        raise ValueError(f"SPECPROD required in yaml file {production_yaml}")
+    specprod = str(conf['SPECPROD']).lower()
+    specprod = verify_variable_with_environment(var=specprod, var_name='specprod',
+                                                env_name='SPECPROD')
+    user = os.environ['USER']
+
+    ## Look for sentinal
+    sentinel_file = os.path.join(os.environ['DESI_SPECTRO_REDUX'],
+                                 os.environ['SPECPROD'], 'run', 'sentinel.txt')
+    if os.path.exists(sentinel_file):
+        log.info(f"Sentinel file {sentinel_file} exists, therefore all "
+                 + f"nights already submitted.")
+        return 0
```

Comment:
One use of this sentintel file is for submit_prod to tell itself that it is already done so that future scron invocations don't keep trying.  Is another potential use to temporarily turn off job submissions without having to edit the scrontab, and then it could resume after the file is removed?

Minor: let's give this a more descriptive name like "done.txt"

Reply:
I had considered that use case, but I agree it could be used for that and it is a clever idea to avoid editing the scrontab. It's similar to the idea behind the yaml file also protecting us from having to change command line arguments mid-prod. 

I've renamed it `prod_submission_complete.txt`, which is very explicit.